### ISSUE TITLE
Support optional finally

### ISSUE THREAD

**Issue by torarnek** [xjamundx/eslint-plugin-promise#29]
It would be great to support the stage 2 proposal of the optional finally. 
[https://github.com/tc39/proposal-promise-finally](https://github.com/tc39/proposal-promise-finally)

This feature is provided with this great shim:
[https://github.com/es-shims/Promise.prototype.finally](https://github.com/es-shims/Promise.prototype.finally)

**Comment by xjamundx** [comment]
What specifically should we support. Basically don't need to `catch` if we have `finally`?

**Comment by torarnek** [comment]
I think the most important feature is that "catch-or-return" should not fail when .finally() is provided after a .catch().

**Comment by xjamundx** [comment]
Love it and should be very doable, any chance you can make a pull request?

Here's a simplified way to detect the `finally`:

```js
export default function(context) {
  return {
    MemberExpression(node) {
      if (node.property && node.property.name === 'then') {
        if (node.parent.parent.type ==='MemberExpression' && node.parent.parent.property.name === 'finally') {
            context.report(node, 'do not use finaly')
		}
      }
    }
  };
};
```

We just need the inverse....

http://astexplorer.net/#/5KDnhNsz9c

**Comment by xjamundx** [comment]
I think this is fixed with the new `terminationMethods` option. We can revisit later if it should be baked in.
